File: lib/src/main/java/me/ameriod/lib/mvp/view/deligate/ActivityDelegate.java
package me.ameriod.lib.mvp.view.deligate;

import android.os.Bundle;
import android.support.annotation.Nullable;

import me.ameriod.lib.mvp.Mvp;

public interface ActivityDelegate<V extends Mvp.View, P extends Mvp.Presenter<V>> {

    void onCreate(@Nullable Bundle savedInstanceState);

    void onSaveInstanceState(@Nullable Bundle outState);

    void onDestroy();

    P getPresenter();
}

File: lib/src/main/java/me/ameriod/lib/mvp/deligate/ActivityDelegate.kt
package me.ameriod.lib.mvp.deligate

import android.os.Bundle

import me.ameriod.lib.mvp.Mvp

interface ActivityDelegate<in V : Mvp.View, out P : Mvp.Presenter<in V>> {

    fun onCreate(savedInstanceState: Bundle?)

    fun onSaveInstanceState(outState: Bundle?)

    fun onDestroy()

    fun getPresenter(): P
}
